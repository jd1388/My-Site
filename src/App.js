import React, { Component } from 'react';

import About from './components/About';
import Experience from './components/Experience';
import Projects from './components/Projects';
import Skills from './components/Skills';
import Footer from './components/Footer';

import Styles from './styles/App';

class App extends Component {
    render() {
        const isMobile = window.innerWidth < 800;

        return (
            <div style={isMobile ? Styles.mobileContent : Styles.content}>
                <About mobile={isMobile}/>
                <Experience mobile={isMobile}/>
                <Projects mobile={isMobile}/>
                <Skills/>
                <Footer/>
            </div>
        );
    }
}

export default App;
