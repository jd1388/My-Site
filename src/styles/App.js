import styled from 'styled-components';

export const AppContainer = styled.div`
    font-size: 12pt;
    overflow: hidden;

    @media (min-width: 800px) {
      font-size: 16pt;
    }
`;
