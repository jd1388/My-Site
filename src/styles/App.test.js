import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { AppContainer } from './App';

describe('App Styles', () => {
    test('renders the AppContainer correctly', () => {
        const wrapper = shallow(<AppContainer />);

        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
