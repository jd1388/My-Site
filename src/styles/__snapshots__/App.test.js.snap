// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Styles renders the AppContainer correctly 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
    font-size: 12pt;
    overflow: hidden;

    @media (min-width: 800px) {
      font-size: 16pt;
    }
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "usesTheme": false,
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;
